pandas/_libs/src/ujson/lib/ultrajsonenc.c

```diff
@@ -1176,16 +1176,15 @@ char *JSON_EncodeObject(JSOBJ obj, JSONObjectEncoder *enc, char *_buffer,
     enc->offset = enc->start;
 
     locale = setlocale(LC_NUMERIC, NULL);
```

Comment:
It turns out setlocale can overwrite this string on subsequent calls, which we are doing with ``setlocale(LC_NUMERIC, "C")``.  I guess the strdup is necessary then(although we might be able to replace with malloc/memcpy).

Reference: https://www.gnu.org/software/libc/manual/html_node/Setting-the-Locale.html#:~:text=The%20string%20returned%20by%20setlocale%20can%20be%20overwritten%20by%20subsequent%20calls%2C%20so%20you%20should%20make%20a%20copy%20of%20the%20string%20(see%20Copying%20Strings%20and%20Arrays)%20if%20you%20want%20to%20save%20it%20past%20any%20further%20calls%20to%20setlocale.%20(The%20standard%20library%20is%20guaranteed%20never%20to%20call%20setlocale%20itself.)

Reply:
Also a nice find. Cool can copy without strdup